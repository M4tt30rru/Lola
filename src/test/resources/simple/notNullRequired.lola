##Import "std.lola"
##Find(SafeArgumentDeclaration) safe ##ArgumentDeclaration
##Find(SafeArgumentDeclarationList) ##NoneOrMore ##Either ##ArgumentDeclaration(_dec) ##or ##SafeArgumentDeclaration(_dec) ##separator ,
##Find ##Type(_type) ##Identifier(_name) (##SafeArgumentDeclarationList(_ls)) {##Any(_statements)}
 ##replace ##(_type) ##(_name) (##(",".join([l._type.name + " " + l._name.name for l in _ls._decs]))) {##("".join(["Objects.requireNonNull(" + l._name.name + ");" for l in _ls._decs]))##(_statements)}

int f(safe Integer x, safe String s) {
 return x + Integer.parse(s);
}
