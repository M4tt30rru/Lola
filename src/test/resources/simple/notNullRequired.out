int f(Integer x, String s) {
 Objects.requireNonNull(x);
 Objects.requireNonNull(s);
 return x + Integer.parse(s);
}
